Font Library: Fix font loading for uploaded fonts with names of more than one word.

juanfra opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Update the use of quotes in font family names, so that they're properly loaded in the browser.

Fixes https://github.com/WordPress/gutenberg/issues/58765

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
When uploading a font with a name of more than one word, `formatFontFamily` was formatting things to in a way that was preventing the font from being loaded correctly in the browser (for a later use). As a result, the font would only be available after refresh.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

Updating the way the font family is being formatted so that it can be loaded correctly in the browser. And adding the font to the whole document so that it's also accessible in the sidebar.

## Testing Instructions

**Uploading new fonts**

1. Download the "Super Enjoy" font from https://www.dafont.com/super-enjoy.font.
2. Upload and activate the font in the Font Library modal.
3. Apply the newly activated font to a heading block in the Gutenberg editor.
4. Observe that the canvas updates to show the new font.

**Installing Google fonts**

1. Install the "Abril Fatface" font from Google fonts.
2. Confirm that the font family has effect on the sidebar.
3. Apply the newly activated font to a heading block in the Gutenberg editor.
4. Observe that the canvas updates to show the new font.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

https://github.com/WordPress/gutenberg/assets/252415/48e2eb69-bfe1-4522-aee8-2d83663bddf7

juanfra committed:
Remove `formatFontFamily` from `loadFontFaceInBrowser` to avoid problems loading fonts with several words in the font name.

matiasbenedetto reviewed (commented):
Hi @juanfra :wave: , thanks for working on this.

I did some testing, and it's working for the font family mentioned in the original issue description. Still, it's not working as expected using other font families with multi-word names:

https://github.com/WordPress/gutenberg/assets/1310626/6ac87413-8ba8-4e81-89d5-7dff4250c208

juanfra committed:
Only wrap multi-word font families with quotes when there are several fonts. Avoid using quotes when it's only one font, so Google Chrome can load it properly.

juanfra committed:
Load font face in the browser so that it can also affect the sidebar.

juanfra committed:
Fix typo in variables.

juanfra commented:
Thanks @matiasbenedetto ✨ 

I've been checking that, and apparently, Google Chrome was picky about loading the font if the font family had quotes in it. I've pushed some changes that should also fix that. It wasn't noticeable before because the font library is showing the svg.

I also added that the font is [added to the browser when installing](https://github.com/WordPress/gutenberg/pull/58853/files#diff-ee8604b8acd83e45973351b79dc6983dc4daad84903a30a41e105a3d7429b74bR405), as the preview in the sidebar wasn't working when installing from Google fonts.

🎥 **This is how it's looking now**

https://github.com/WordPress/gutenberg/assets/252415/6be57534-f6bb-4ce0-ba3b-6f2f9fa08bb0

(I also fixed some typos in a couple of variables)

juanfra committed:
Update test.

matiasbenedetto commented:
@juanfra thanks again for working on this.

I have been debugging the problem mentioned here https://github.com/WordPress/gutenberg/pull/58853#issuecomment-1939115222 and I found several edge cases (Wordpress theme.json slugs formatting, firefox specific needs, and differences between CSS font-family value and name of font faces loaded by JS) so I submitted a PR alternative to this one which I think is (hopefully) covering most of the cases: https://github.com/WordPress/gutenberg/pull/59019

Could you please review and test it?
